src/quantum/azext_quantum/operations/job.py

```diff
@@ -0,0 +1,238 @@
+
+    args.extend(program_args)
+
+    logger.debug("Running project with arguments:")
+    logger.debug(args)
+
+    return args
+
+
+def submit(cmd, program_args, resource_group_name=None, workspace_name=None, target_id=None, project=None,
+           job_name=None, shots=None, no_build=False):
+    """
+    Submits a Q# project for execution to Azure Quantum.
+    """
+
+    # We first build and then call run.
+    # Can't call run directly because it fails to understand the
+    # `ExecutionTarget` property when passed in the command line
+    if not no_build:
+        build(cmd, target_id=target_id, project=project)
+
+    logger.info("Project built successfully.")
+
+    ws = WorkspaceInfo(cmd, resource_group_name, workspace_name)
+    target = TargetInfo(cmd, target_id)
+    token = _get_data_credentials(cmd.cli_ctx, ws.subscription).get_token().token
+
+    args = _generate_submit_args(program_args, ws, target, token, project, job_name, shots)
+
+    import subprocess
+    result = subprocess.run(args, stdout=subprocess.PIPE, check=False)
+
+    if result.returncode == 0:
+        job_id = result.stdout.decode('ascii').strip()
+        return show(cmd, job_id, resource_group_name, workspace_name)
+
+    raise CLIError("Failed to submit job.")
+
+
+def _parse_blob_url(url):
+    from urllib.parse import urlparse
+    o = urlparse(url)
+
+    account_name = o.netloc.split('.')[0]
+    container = o.path.split('/')[-2]
+    blob = o.path.split('/')[-1]
+    sas_token = o.query
+
+    return {
+        "account_name": account_name,
+        "container": container,
+        "blob": blob,
+        "sas_token": sas_token
+    }
+
+
+def output(cmd, job_id, resource_group_name=None, workspace_name=None):
+    """ Returns back the results of a Q# execution """
+    import tempfile
+    import json
+    import os
+    from azure.cli.command_modules.storage._client_factory import blob_data_service_factory
+
+    path = os.path.join(tempfile.gettempdir(), job_id)
+
+    if os.path.exists(path):
+        logger.debug("Using existing blob from %s", path)
+    else:
+        logger.debug("Downloading job results blob into %s", path)
+
+        info = WorkspaceInfo(cmd, resource_group_name, workspace_name)
+        client = cf_jobs(cmd.cli_ctx, info.subscription, info.resource_group, info.name)
+        job = client.get(job_id)
+
+        if job.status != "Succeeded":
+            return f"Job status: {job.status}. Output only available if Succeeded."
+
+        args = _parse_blob_url(job.output_data_uri)
+        blob_service = blob_data_service_factory(cmd.cli_ctx, args)
+        blob_service.get_blob_to_path(args['container'], args['blob'], path)
+
+    with open(path) as json_file:
+        data = json.load(json_file)
+        return data
+
+
+def wait(cmd, job_id, resource_group_name=None, workspace_name=None, max_poll_wait_secs=5):
+    """
+    Place the CLI in a waiting state until the job finishes execution.
+    """
+    import time
```

Comment:
You can reference `ARMPolling` [here](https://github.com/Azure/msrestazure-for-python/blob/698eb06bcbf9fda4c7ae04d929f7b5b955fcbfed/msrestazure/polling/arm_polling.py#L340). And is there any special logic for this ? If no, maybe we can just use the default LongRunningOperation.

Reply:
The data-plane has not implemented yet ARM's [async operation status](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/async-operations), as such I believe we can't use `ARMPolling` or `LongRunningOperation`.